apps/studio/src/components/Intercom.tsx

```diff
@@ -0,0 +1,24 @@
+import { Intercom as IntercomSDK } from "@intercom/messenger-js-sdk"
+
+import { env } from "~/env.mjs"
+import { useMe } from "~/features/me/api"
+
+const convertDateToUnixTimestamp = (date: Date): number => {
+  return Math.floor(date.getTime() / 1000)
+}
+
+export const Intercom = () => {
+  const { me } = useMe()
+
+  if (env.NEXT_PUBLIC_INTERCOM_APP_ID) {
+    IntercomSDK({
+      app_id: env.NEXT_PUBLIC_INTERCOM_APP_ID,
+      user_id: me.id,
+      name: me.name,
```

Comment:
issue: I think we should be using `preferredName` here instead? Also, the `name` column can be empty, so we should fallback to the front part of the email address with replacements to spaces if it is not present.

Reply:
actually, on this, why is there a need for both `preferredName` and `name` columns?

Also, i've looked at the DB, `name` are all filled but not `preferredName`, so it's safer to use `name` instead

Noted on the fallback on email, will do!

Reply:
Hmm the `preferredName` column was created specifically for Intercom actually, iirc it is to make the widget more personal to the user. I think it's just very low on our priority list to allow users to define their preferred names in Studio.

Reply:
so intended use case was for name to be `Adrian Goh Jun Wei` and preferred name be `Adrian`?

Reply:
Yup, so the Intercom widget will say "Hi, Adrian" rather than the full name

Reply:
okay lemme fallback from `preferredName` > `name` > `email` username